src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6144,6 +6142,7 @@ void VM_Crac::read_shm(int shmid) {
 
     shm_unlink(shmpath);
 
+    delete _restore_parameters;
     _restore_parameters = CracRestoreParameters::read_from(shmfd);
```

Comment:
`CracRestoreParameters::read_from` can return NULL.  If we need to ensure `_restore_parameters` is not null, then we need to do something like this instead:

```suggestion
    CracRestoreParameters *original_parameters = _restore_parameters;
    _restore_parameters = CracRestoreParameters::read_from(shmfd);
    if (_restore_parameters == NULL) {
      _restore_parameters = original_parameters;
    } else {
      delete original_parameters;
   }
```

Reply:
Indeed, thanks! Fixed